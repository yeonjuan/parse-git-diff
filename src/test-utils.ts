import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
