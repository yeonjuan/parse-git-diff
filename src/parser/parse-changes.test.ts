import { createContext } from '../test-utils';
import parseChanges from './parse-changes';

describe('parseChanges', () => {
  it('should parse changes.', () => {
    // prettier-ignore
    const src =
` unchanged
 -deleted line
 +added line
 unchanged`;

    const result = parseChanges(createContext(src));

    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });
});
