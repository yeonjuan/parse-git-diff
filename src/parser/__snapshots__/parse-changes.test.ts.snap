// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseChanges should parse changes. 1`] = `
Array [
  Object {
    "content": "unchanged",
    "line": 1,
    "type": "Unchanged",
  },
  Object {
    "content": "-deleted line",
    "line": 2,
    "type": "Unchanged",
  },
  Object {
    "content": "+added line",
    "line": 3,
    "type": "Unchanged",
  },
  Object {
    "content": "unchanged",
    "line": 4,
    "type": "Unchanged",
  },
]
`;
